solutions/usaco-839.mdx

```diff
@@ -0,0 +1,56 @@
+---
+id: usaco-839
+source: USACO Gold 2018 Open
+title: Talent Show
+author: Arpan Banerjee
+---
+
+[Official Editorial](http://www.usaco.org/current/data/sol_talent_gold_open18.html)
+
+## Incorrect Solution — Common Misconception
+Instead of using binary search as the official solution does, a knapsack can be done. An array of talents can be kept, where `talent[i]` is the minimum weight required to create a talent sum of i. To make sure each cow is used at most once and the order of the cows doesn't matter, the indices of `talent` are iterated though from greatest to least, and the `cows` are iterated though in the outer loop; the implementation should make this clear. This maximizes the ratio $t/w$ as desired. Then, the maximum ratio of $t_i / w_i$ can be easily found by going through `talent` and ignoring any $w_i$ that is less than $W$. 
+
+My $\mathcal{O}(250 \cdot 10^3 \cdot n)$ code is below:
```

Comment:
err something like O(MAXT \cdot N) would be better (mention that MAXT is max talent as well)